serialization.rst

```diff
@@ -49,6 +49,19 @@ and should throw an ``MongoDB\Driver\Exception\UnexpectedValueException``
 exception. It is valid to return a packed array, but it must also be stored as
 a **BSON document**.
 
+If an object is of a class that implements the
+``MongoDB\BSON\ArraySerializable`` interface, call ``bsonSerialize`` and store
+the *values* of the returned array as elements of a **BSON array**. Not
+returning an array is an error and should throw an
+``MongoDB\Driver\Exception\UnexpectedValueException`` exception. It is valid to
+return an associative array, but the keys will be ignored and the array elements
+will be re-indexed (akin to filtering the array with ``array_values()``).
```

Comment:
I think userland should make the choice themselves - no unexpected data loss that way. Which means thrown an exception if it's not a packed array.